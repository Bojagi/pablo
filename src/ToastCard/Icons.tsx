import React from 'react';

export const ErrorIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path fill="none" d="M0 0h24v24H0z" />
    <path d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-1-5h2v2h-2v-2zm0-8h2v6h-2V7z" />
  </svg>
);

export const SuccessIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path fill="none" d="M0 0h24v24H0z" />
    <path d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-.997-4L6.76 11.757l1.414-1.414 2.829 2.829 5.656-5.657 1.415 1.414L11.003 16z" />
  </svg>
);

export const WarningIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path fill="none" d="M0 0h24v24H0z" />
    <path d="M12.866 3l9.526 16.5a1 1 0 0 1-.866 1.5H2.474a1 1 0 0 1-.866-1.5L11.134 3a1 1 0 0 1 1.732 0zm-8.66 16h15.588L12 5.5 4.206 19zM11 16h2v2h-2v-2zm0-7h2v5h-2V9z" />
  </svg>
);

export const InfoIcon = () => (
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24">
    <path fill="none" d="M0 0h24v24H0z" />
    <path d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zM11 7h2v2h-2V7zm0 4h2v6h-2v-6z" />
  </svg>
);
