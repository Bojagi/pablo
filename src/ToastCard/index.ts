import './styles';

export * from './ToastCard';
