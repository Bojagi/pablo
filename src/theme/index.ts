export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
