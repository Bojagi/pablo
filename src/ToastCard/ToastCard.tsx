import React, { ReactElement, ReactNode } from 'react';
import styled, { css } from 'styled-components';
import { Flex, Box, layoutInterpolationFn, LayoutBoxProps } from '../Box';
import { IconButton } from '../IconButton';
import { CloseIcon } from '../Icons';
import { useComponentStyle } from '../theme';
import { Paragraph, Subtitle } from '../Typography';
import { getColor, getComponentStyle, shadowTransformer } from '../utils/styleHelpers';
import { ErrorIcon, InfoIcon, SuccessIcon, WarningIcon } from './Icons';

const CardWrapper = styled.div<LayoutBoxProps>`
  border-radius: ${getComponentStyle('toastCard.borderRadius')};
  max-width: ${getComponentStyle('toastCard.width')};
  box-sizing: border-box;
  background-color: ${getComponentStyle('toastCard.backgroundColor')};
  color: ${getComponentStyle('toastCard.color')};
  padding: ${getComponentStyle('toastCard.padding')};
  box-shadow: ${getComponentStyle('toastCard.shadow', shadowTransformer)};
  ${layoutInterpolationFn};
`;

export type ToastType = 'info' | 'success' | 'warning' | 'error';

export interface ToastCardProps extends LayoutBoxProps {
  title: ReactNode;
  description?: ReactNode;
  type?: ToastType;
  icon?: ReactNode;
  closable?: boolean;
  onClose?: () => void;
}

const ALLOWED_TYPES: ToastType[] = ['info', 'success', 'warning', 'error'];

const iconColors: Record<ToastType, string> = {
  info: 'common.white',
  success: 'positive.main',
  warning: 'neutral.main',
  error: 'negative.main',
};

const icons: Record<ToastType, ReactElement> = {
  info: <InfoIcon />,
  success: <SuccessIcon />,
  warning: <WarningIcon />,
  error: <ErrorIcon />,
};

const displayBlockIconsCss = css<LayoutBoxProps>`
  & > * {
    display: block;
  }
`;

export function ToastCard({
  title,
  description,
  type: typeProp,
  icon,
  closable = false,
  onClose = () => {},
  ...props
}: ToastCardProps) {
  const type = typeProp === undefined || ALLOWED_TYPES.includes(typeProp) ? typeProp : 'info';
  const color = useComponentStyle('toastCard.color') as string;

  return (
    <CardWrapper data-testid="pbl-toastcard" {...props}>
      <Flex alignItems={description ? 'flex-start' : 'center'}>
        {(type || icon) && (
          <Box
            mr={5}
            // display="inline-block"
            data-testid="pbl-toastcard-iconbox"
            fillColor={iconColors[type || 'info']}
            css={displayBlockIconsCss}
          >
            {icon || icons[type!]}
          </Box>
        )}
        <Box flexGrow={1}>
          <Flex>
            <Subtitle data-testid="pbl-toastcard-title" flexGrow={1} inline={!description}>
              {title}
            </Subtitle>
            {closable && (
              <IconButton
                data-testid="pbl-toastcard-closebtn"
                mx={-5}
                onClick={onClose}
                size="small"
                css={
                  css`
                    fill: ${color};
                    &:hover:not(:disabled) {
                      background-color: ${getColor('blackOpacity', '300')};
                    }
                  ` as any
                }
              >
                <CloseIcon />
              </IconButton>
            )}
          </Flex>
          {description && (
            <Paragraph inline data-testid="pbl-toastcard-description">
              {description}
            </Paragraph>
          )}
        </Box>
      </Flex>
    </CardWrapper>
  );
}
