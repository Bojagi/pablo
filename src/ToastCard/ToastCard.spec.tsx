import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';
import { defaultTheme, PabloThemeProvider } from '../theme';
import { ToastCard } from './ToastCard';

test('Render toast card with no type and with only title', () => {
  const { container } = renderComponent({ title: 'Hi' });
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild?.textContent).toBe('Hi');
});

test('Render toast card with title and description', () => {
  const { getByTestId } = renderComponent({
    title: 'Hi',
    description: 'Something happened!',
  });
  expect(getByTestId('pbl-toastcard-description')).toHaveTextContent('Something happened!');
});

test('Render closable toast card', () => {
  const onCloseMock = jest.fn();
  const { getByTestId } = renderComponent({
    title: 'Hi',
    onClose: onCloseMock,
    description: 'Something happened!',
    closable: true,
  });

  expect(onCloseMock).toHaveBeenCalledTimes(0);
  act(() => {
    fireEvent.click(getByTestId('pbl-toastcard-closebtn'));
  });
  expect(onCloseMock).toHaveBeenCalledTimes(1);
});

describe.each([
  [undefined, defaultTheme.colors.common.white, { icon: <div>hi</div> }],
  ['info', defaultTheme.colors.common.white, {}],
  ['success', defaultTheme.colors.positive.main, {}],
  ['warning', defaultTheme.colors.neutral.main, {}],
  ['error', defaultTheme.colors.negative.main, {}],
  ['unknown', defaultTheme.colors.common.white, { icon: <div>hi</div> }],
])('type %s', (type, expectedIconColor, additionalProps) => {
  test('Render toast card with only title', () => {
    const { container, getByTestId } = renderComponent({ type, title: 'Hi', ...additionalProps });
    expect(container.firstChild).toMatchSnapshot();
    expect(getByTestId('pbl-toastcard-iconbox')).toHaveStyleRule('fill', expectedIconColor);
  });

  test('Render toast card with custom icon', () => {
    const { getByTestId } = renderComponent({ type, title: 'Hi', icon: <div>hello</div> });
    expect(getByTestId('pbl-toastcard-iconbox')).toMatchSnapshot();
    expect(getByTestId('pbl-toastcard-iconbox')).toHaveStyleRule('fill', expectedIconColor);
    expect(getByTestId('pbl-toastcard-iconbox')).toHaveTextContent('hello');
  });
});

function renderComponent(props) {
  return render(
    <PabloThemeProvider>
      <ToastCard {...props} />
    </PabloThemeProvider>
  );
}
