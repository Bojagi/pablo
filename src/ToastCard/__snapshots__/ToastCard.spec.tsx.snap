// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render toast card with no type and with only title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-radius: 6px;
  max-width: 400px;
  background-color: var(--pbl-theme-colors-gray-800);
  color: var(--pbl-theme-colors-common-blackContrastText);
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(1.5rem, 1.2857142857142856rem + 1.0714285714285714vw, 2.25rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  aria-describedby=":r0:-title"
  aria-labelledby=":r0:-title"
  class="emotion-0"
  data-testid="pbl-toastcard"
  id=":r0:"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
    >
      <div
        class="emotion-3"
      >
        <h4
          class="emotion-4"
          data-testid="pbl-toastcard-title"
          id=":r0:-title"
        >
          Hi
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`type error Render toast card with custom icon 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-negative-main);
}

.emotion-0>* {
  display: block;
}

.emotion-0>* {
  display: block;
}

<div
  class="emotion-0"
  data-testid="pbl-toastcard-iconbox"
>
  <div>
    hello
  </div>
</div>
`;

exports[`type error Render toast card with only title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-radius: 6px;
  max-width: 400px;
  background-color: var(--pbl-theme-colors-gray-800);
  color: var(--pbl-theme-colors-common-blackContrastText);
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(1.5rem, 1.2857142857142856rem + 1.0714285714285714vw, 2.25rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-negative-main);
}

.emotion-2>* {
  display: block;
}

.emotion-2>* {
  display: block;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  aria-describedby=":rb:-title"
  aria-labelledby=":rb:-title"
  class="emotion-0"
  data-testid="pbl-toastcard"
  id=":rb:"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
      data-testid="pbl-toastcard-iconbox"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-1-5h2v2h-2v-2zm0-8h2v6h-2V7z"
        />
      </svg>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        <h4
          class="emotion-5"
          data-testid="pbl-toastcard-title"
          id=":rb:-title"
        >
          Hi
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`type info Render toast card with custom icon 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-common-white);
}

.emotion-0>* {
  display: block;
}

.emotion-0>* {
  display: block;
}

<div
  class="emotion-0"
  data-testid="pbl-toastcard-iconbox"
>
  <div>
    hello
  </div>
</div>
`;

exports[`type info Render toast card with only title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-radius: 6px;
  max-width: 400px;
  background-color: var(--pbl-theme-colors-gray-800);
  color: var(--pbl-theme-colors-common-blackContrastText);
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(1.5rem, 1.2857142857142856rem + 1.0714285714285714vw, 2.25rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-common-white);
}

.emotion-2>* {
  display: block;
}

.emotion-2>* {
  display: block;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  aria-describedby=":r5:-title"
  aria-labelledby=":r5:-title"
  class="emotion-0"
  data-testid="pbl-toastcard"
  id=":r5:"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
      data-testid="pbl-toastcard-iconbox"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zM11 7h2v2h-2V7zm0 4h2v6h-2v-6z"
        />
      </svg>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        <h4
          class="emotion-5"
          data-testid="pbl-toastcard-title"
          id=":r5:-title"
        >
          Hi
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`type success Render toast card with custom icon 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-positive-main);
}

.emotion-0>* {
  display: block;
}

.emotion-0>* {
  display: block;
}

<div
  class="emotion-0"
  data-testid="pbl-toastcard-iconbox"
>
  <div>
    hello
  </div>
</div>
`;

exports[`type success Render toast card with only title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-radius: 6px;
  max-width: 400px;
  background-color: var(--pbl-theme-colors-gray-800);
  color: var(--pbl-theme-colors-common-blackContrastText);
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(1.5rem, 1.2857142857142856rem + 1.0714285714285714vw, 2.25rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-positive-main);
}

.emotion-2>* {
  display: block;
}

.emotion-2>* {
  display: block;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  aria-describedby=":r7:-title"
  aria-labelledby=":r7:-title"
  class="emotion-0"
  data-testid="pbl-toastcard"
  id=":r7:"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
      data-testid="pbl-toastcard-iconbox"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm0-2a8 8 0 1 0 0-16 8 8 0 0 0 0 16zm-.997-4L6.76 11.757l1.414-1.414 2.829 2.829 5.656-5.657 1.415 1.414L11.003 16z"
        />
      </svg>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        <h4
          class="emotion-5"
          data-testid="pbl-toastcard-title"
          id=":r7:-title"
        >
          Hi
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`type undefined Render toast card with custom icon 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-common-white);
}

.emotion-0>* {
  display: block;
}

.emotion-0>* {
  display: block;
}

<div
  class="emotion-0"
  data-testid="pbl-toastcard-iconbox"
>
  <div>
    hello
  </div>
</div>
`;

exports[`type undefined Render toast card with only title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-radius: 6px;
  max-width: 400px;
  background-color: var(--pbl-theme-colors-gray-800);
  color: var(--pbl-theme-colors-common-blackContrastText);
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(1.5rem, 1.2857142857142856rem + 1.0714285714285714vw, 2.25rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-common-white);
}

.emotion-2>* {
  display: block;
}

.emotion-2>* {
  display: block;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  aria-describedby=":r3:-title"
  aria-labelledby=":r3:-title"
  class="emotion-0"
  data-testid="pbl-toastcard"
  id=":r3:"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
      data-testid="pbl-toastcard-iconbox"
    >
      <div>
        hi
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        <h4
          class="emotion-5"
          data-testid="pbl-toastcard-title"
          id=":r3:-title"
        >
          Hi
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`type unknown Render toast card with custom icon 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-common-white);
}

.emotion-0>* {
  display: block;
}

.emotion-0>* {
  display: block;
}

<div
  class="emotion-0"
  data-testid="pbl-toastcard-iconbox"
>
  <div>
    hello
  </div>
</div>
`;

exports[`type unknown Render toast card with only title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-radius: 6px;
  max-width: 400px;
  background-color: var(--pbl-theme-colors-gray-800);
  color: var(--pbl-theme-colors-common-blackContrastText);
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(1.5rem, 1.2857142857142856rem + 1.0714285714285714vw, 2.25rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-common-white);
}

.emotion-2>* {
  display: block;
}

.emotion-2>* {
  display: block;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  aria-describedby=":rd:-title"
  aria-labelledby=":rd:-title"
  class="emotion-0"
  data-testid="pbl-toastcard"
  id=":rd:"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
      data-testid="pbl-toastcard-iconbox"
    >
      <div>
        hi
      </div>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        <h4
          class="emotion-5"
          data-testid="pbl-toastcard-title"
          id=":rd:-title"
        >
          Hi
        </h4>
      </div>
    </div>
  </div>
</div>
`;

exports[`type warning Render toast card with custom icon 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-neutral-main);
}

.emotion-0>* {
  display: block;
}

.emotion-0>* {
  display: block;
}

<div
  class="emotion-0"
  data-testid="pbl-toastcard-iconbox"
>
  <div>
    hello
  </div>
</div>
`;

exports[`type warning Render toast card with only title 1`] = `
.emotion-0 {
  box-sizing: border-box;
  border-radius: 6px;
  max-width: 400px;
  background-color: var(--pbl-theme-colors-gray-800);
  color: var(--pbl-theme-colors-common-blackContrastText);
  padding: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(1.5rem, 1.2857142857142856rem + 1.0714285714285714vw, 2.25rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem) clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  box-shadow: 0px 4px 20px rgba(0, 0, 0, 0.2);
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  box-sizing: border-box;
  margin-right: clamp(0.75rem, 0.6428571428571428rem + 0.5357142857142857vw, 1.125rem);
  fill: var(--pbl-theme-colors-neutral-main);
}

.emotion-2>* {
  display: block;
}

.emotion-2>* {
  display: block;
}

.emotion-3 {
  box-sizing: border-box;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-5 {
  box-sizing: border-box;
  font-family: var(--pbl-theme-typography-base-fontFamily);
  font-weight: var(--pbl-theme-typography-base-fontWeight);
  font-style: normal;
  margin: 0;
  line-height: var(--pbl-theme-typography-h4-lineHeight);
  font-size: var(--pbl-theme-typography-h4-fontSize);
  margin-bottom: var(--pbl-theme-typography-h4-marginBottom);
  font-weight: var(--pbl-theme-typography-h4-fontWeight);
  margin-bottom: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  aria-describedby=":r9:-title"
  aria-labelledby=":r9:-title"
  class="emotion-0"
  data-testid="pbl-toastcard"
  id=":r9:"
>
  <div
    class="emotion-1"
  >
    <div
      class="emotion-2"
      data-testid="pbl-toastcard-iconbox"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12.866 3l9.526 16.5a1 1 0 0 1-.866 1.5H2.474a1 1 0 0 1-.866-1.5L11.134 3a1 1 0 0 1 1.732 0zm-8.66 16h15.588L12 5.5 4.206 19zM11 16h2v2h-2v-2zm0-7h2v5h-2V9z"
        />
      </svg>
    </div>
    <div
      class="emotion-3"
    >
      <div
        class="emotion-4"
      >
        <h4
          class="emotion-5"
          data-testid="pbl-toastcard-title"
          id=":r9:-title"
        >
          Hi
        </h4>
      </div>
    </div>
  </div>
</div>
`;
